DOC-2284: Add new documentation for `Revision History` Plugin. (#3071)
tinymce/tinymce-docs: modules/ROOT/partials/configuration/revisionhistory_fetch.adoc

[[revisionhistory_fetch]]
== `revisionhistory_fetch`

The plugin uses the `revisionhistory_fetch` asynchronous function to retreive the revision history list.

The `revisionhistory_fetch` option is a function that returns a https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise[Promise] that resolves to an array of objects representing the revision history.

[NOTE]
The revision history plugin does not automatically sort results before displaying. It is the responsibility of the user to sort the results according to their preferences, such as sorting them in reverse chronological order based on the `createdAt` date.

////

> What does the option do?

Author: anything else to add here? @tiny-ben-tran

    createdAt: '2023-11-25T08:30:21.578Z',
    content: `
      <p>Welcome to TinyMCE Docs!</p>
      <p>Here is some content that is bold and italic.</p>
    `,
  },
  {
    revisionId: '1',
    createdAt: '2023-11-24T22:26:21.578Z',
    content: `
      <p>Welcome to Tinymce!</p>
    `,
  },
]
----

The data returned by `revisionhistory_fetch` must adhere to the following requirements:

. Each list item must have a unique `revisionId` string value.
. Each list item must have a `createdAt` string value representing a datetime in ISO-8061 format.
. Each list item must have a `content` string value representing the HTML content of the revision.

=== Example: using `revisionhistory_fetch`

[source,js]
----
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML
	plugins: 'revisionhistory',
  toolbar: 'revisionhistory',
  revisionhistory_fetch: () =>
    fetch('/revisions', {
      method: 'GET',
    })
    .then((response) => response.json())
    .then((data) => data)
    .catch((error) => console.log('Failed to get revision history\n' + error)),
});
----

// Remove if not applicable.
// Edit the sub-head to singular or plural as required.

> === Limitation(s) of the `revisionhistory_fetch` option

Author: limitations of `revisionhistory_fetch`? @tiny-ben-tran

[[revisionhistory_fetch]]
== `revisionhistory_fetch`

The plugin uses the `revisionhistory_fetch` function to retrieve the saved versions.

The `revisionhistory_fetch` option is an asynchronous function that returns a https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise[Promise] which resolves to an array of revisions.

[NOTE]
The {pluginname} plugin does not sort the result before displaying. This allows the integrators to sort the result according to their preference. It is recommended that the result should be sorted in reverse chronological order because it is more intuitive for the users when they can only identify a version by its date.

*Type:* `+Function+` (Returns a https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise[Promise])

> *Input parameters:*
> None

Reviewer: ```suggestion
*Input parameters:* None
```

*Return data:* `+Array+`

=== Example of `revisionhistory_fetch` response

[source,js]
----
const revisions = [
  {
    revisionId: '3',
    createdAt: '2023-11-29T10:11:21.578Z',
    content: `
      <h2>Welcome to TinyMCE Docs!</h2>
      <p>Here is some content that is <strong>bold</strong> and <em>italic</em>.</p>
    `,
  },
  {
    revisionId: '2',
    createdAt: '2023-11-25T08:30:21.578Z',
    content: `
      <p>Welcome to TinyMCE Docs!</p>
      <p>Here is some content that is bold and italic.</p>
    `,
  },
  {
    revisionId: '1',
    createdAt: '2023-11-24T22:26:21.578Z',
    content: `
      <p>Welcome to Tinymce!</p>
    `,
  }
]
----

> The data returned by `revisionhistory_fetch` must have the following fields for the {pluginname} to work properly:

Reviewer: ```suggestion
The data returned by `revisionhistory_fetch` requires the following fields for the {pluginname} to work:
```
